hail/python/hail/ir/ir.py

```diff
@@ -2494,6 +2494,12 @@ def delete(env, name):
         return ApplyAggOp(ir.agg_op,
                           subst_init_op_args,
                           subst_seq_op_args)
+    elif isinstance(ir, AggArrayPerElement):
+        return AggArrayPerElement(_subst(ir.array, delete(agg_env, ir.element_name)),
```

Comment:
this subst shouldn't delete anything, I think. The `array` uses the aggregation environment, but AggArrayPerElement only binds index_name in the body.

Reply:
we use UIDs everywhere so it's kinda possible that this can never cause trigger an error, but should fix it